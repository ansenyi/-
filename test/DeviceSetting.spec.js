/* eslint-disable no-undef,no-unused-vars */
import 'babel-polyfill'
import { mount, createLocalVue } from 'vue-test-utils'
import DeviceSetting from '@/components/Device/DeviceSetting.vue'
import moxios from 'moxios'
import axios from 'axios'
import vueAxios from 'vue-axios'

describe('device setting', () => {
  const localVue = createLocalVue()
  localVue.use(vueAxios, axios)
  it('init and controlCallBack', async() => {
    moxios.install()
    moxios.stubRequest('getDevice', {
      status: 200,
      response: {
        status: true,
        data: {
          deviceName: 'device',
          wiegand: 1,
          outputType: 2,
          matchType: 2,
          delayTime: 4,
          protectLock: 0,
          outputMsg: '',
          softwareVersion: '2.0',
          privilege: 1
        }
      }
    })
    const cmp = mount(DeviceSetting, {
      localVue,
      propsData: {
        deviceId: 'a'
      }
    })
    await cmp.vm.$nextTick()
    await moxios.wait()
    expect(cmp.element).toMatchSnapshot()
    moxios.stubRequest('terminalController', {
      status: 200,
      response: {
        status: true,
        data: [
          {'a': 0}
        ]
      }
    })
    cmp.vm.controlCallBack('强制升级', 3, '1.1')()
    await moxios.wait()
    await cmp.vm.$nextTick()
    expect(cmp.vm.softwareVersion = '1.1')
    expect(cmp.vm.msg[0]).toEqual({detail: '版本回退：1.1', status: '成功'})
    console.log('end controlCallBack')
    moxios.uninstall()
  })
  it('save', async() => {
    moxios.install()
    moxios.stubRequest('getDevice', {
      status: 200,
      response: {
        status: true,
        data: {
          deviceName: 'device',
          wiegand: 1,
          outputType: 2,
          matchType: 2,
          delayTime: 4,
          protectLock: 0,
          outputMsg: '',
          softwareVersion: '2.0',
          privilege: 1
        }
      }
    })
    const cmp = mount(DeviceSetting, {
      localVue,
      propsData: {
        deviceId: 'a'
      }
    })
    await cmp.vm.$nextTick()
    await moxios.wait()
    cmp.vm.deviceName = 'device1'
    cmp.vm.wiegand = '0'
    cmp.vm.outputType = '0'
    cmp.vm.saveEditHandler()
    // cmp.find('.save-back-button').find('.button').trigger('click')
    expect(cmp.find('label[for=out]').find('input:last-of-type').hasClass('red')).toBeTruthy()
    moxios.stubRequest('editDevice', {
      status: 200,
      response: {
        status: true,
        data: [
          {'a': 0}
        ]
      }
    })
    cmp.vm.output = 'msg'
    cmp.vm.protectLock = '1'
    cmp.vm.saveEditHandler()
    // cmp.find('.save-back-button').find('.button').trigger('click')
    await moxios.wait()
    await cmp.vm.$nextTick()
    // expect(cmp.vm.redAdd).toBe(false)
    expect(cmp.vm.initial.deviceName).toBe('device1')
    console.log('end save')
  })
})
