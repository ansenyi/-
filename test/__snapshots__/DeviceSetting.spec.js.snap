// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`device setting init and controlCallBack 1`] = `
<div
  class="device-setting"
>
  <form
    class="form-container"
  >
    <h4
      class="form-title"
    >
      设备参数
    </h4>
     
    <label>
      
      设备编号：
      
      <input
        placeholder="a"
        readonly="readonly"
        type="text"
      />
    </label>
     
    <label>
      
      设备名称：
      
      <input
        autocomplete="off"
        type="text"
      />
    </label>
     
    <label
      class="select-label"
    >
      
      通讯协议：
      
      <select>
        <option
          disabled="disabled"
          selected="selected"
          style="display: none;"
          value="请选择"
        >
          请选择
        </option>
         
        <option
          value="0"
        >
          韦根 26
        </option>
         
        <option
          value="1"
        >
          韦根 34
        </option>
      </select>
       
      <i
        class="select-down icon-down"
      />
    </label>
     
    <label
      class="select-label"
      for="out"
      id="output"
    >
      
      输出类型：
      
      <select
        id="out"
      >
        <option
          disabled="disabled"
          selected="selected"
          style="display: none;"
          value="请选择"
        >
          请选择
        </option>
         
        <option
          value="1"
        >
          工号
        </option>
         
        <option
          value="2"
        >
          卡号
        </option>
         
        <option
          value="0"
        >
          自定义
        </option>
      </select>
       
      <i
        class="select-down icon-down"
      />
       
      <input
        disabled="disabled"
        type="text"
      />
       
      <input
        autocomplete="off"
        style="display: none;"
        type="text"
      />
    </label>
     
    <label
      class="select-label"
      for="check"
    >
      
      比对方式：
      
      <select
        id="check"
      >
        <option
          disabled="disabled"
          selected="selected"
          style="display: none;"
          value="请选择"
        >
          请选择
        </option>
         
        <option
          value="0"
        >
          虹膜对比
        </option>
         
        <option
          value="1"
        >
          卡与虹膜对比
        </option>
         
        <option
          value="2"
        >
          卡或虹膜对比
        </option>
      </select>
       
      <i
        class="select-down icon-down"
      />
    </label>
     
    <label
      class="select-label"
      for="delay"
    >
      
      门禁延时：
      
      <select
        id="delay"
      >
        <option
          disabled="disabled"
          selected="selected"
          style="display: none;"
          value="请选择"
        >
          请选择
        </option>
         
        <option
          value="1000"
        >
          1
        </option>
        <option
          value="3000"
        >
          3
        </option>
        <option
          value="5000"
        >
          5
        </option>
        <option
          value="10000"
        >
          10
        </option>
        <option
          value="20000"
        >
          20
        </option>
      </select>
       
      <i
        class="select-down icon-down"
      />
    </label>
     
    <label
      class="select-label"
      for="protectLock"
    >
      
      防拆开关：
      
      <select
        id="protectLock"
      >
        <option
          disabled="disabled"
          selected="selected"
          style="display: none;"
          value="请选择"
        >
          请选择
        </option>
         
        <option
          value="1"
        >
          开
        </option>
         
        <option
          value="0"
        >
          关
        </option>
      </select>
       
      <i
        class="select-down icon-down"
      />
    </label>
     
    <div
      class="save-back-button"
    >
      <button
        class="button"
      >
        保存
      </button>
       
      <button
        class="button"
      >
        返回
      </button>
    </div>
  </form>
   
  <div
    class="buttons"
  >
    <button
      class="button"
    >
      关机
    </button>
     
    <button
      class="button"
    >
      重启
    </button>
     
    <button
      class="button"
    >
      恢复出厂设置
    </button>
     
    <button
      class="button"
    >
      强制升级
    </button>
     
    <button
      class="button"
    >
      版本回退
    </button>
     
    <button
      class="button"
    >
      强制开门
    </button>
  </div>
   
  <div
    class="console-container"
  >
    <h4
      class="console-title"
    >
      操作记录
    </h4>
     
    <div
      class="console-content"
    />
  </div>
</div>
`;
